Revert-tvOS/Sources/CollectionDataSource.swift

```diff
@@ -1,60 +1,71 @@
-//
-//  Copyright © 2016 Itty Bitty Apps. All rights reserved.
+// Copyright © 2016 Itty Bitty Apps. All rights reserved.
 
-import UIKit
+import Foundation
 
-final class CollectionDataSource<Object: Collectable, Cell: UICollectionViewCell>: NSObject, UICollectionViewDataSource {
-  typealias CellConfigurator = (HomeCollectionCell, _ item: HomeItem) -> Void
-  typealias ItemFilterClosure = (HomeItem) -> Bool
-  typealias GroupFilterClosure = (CollectableGroup<HomeItem>) -> Bool
+final class CollectionDataSource<Section: SectionItem, Cell: UICollectionViewCell>: NSObject, UICollectionViewDataSource {
+  typealias CellConfigurator = (Cell, Section.item) -> Void
+  typealias GroupFilterClosure = ((Section) -> Bool)
+  typealias ItemFilterClosure = (Section.item) -> Bool
 
-  required init(collection: CollectableCollection<HomeItem>, configureCell: @escaping CellConfigurator, cellIdentifier: String) {
-    self.unfilteredCollection = collection
-    self.collection = collection
+  // MARK: - Private properties
+
+  private let unfilteredSections: [Section]
+  private var sections: [Section]
+  private let cellIdentifier: String
+  private let configureCell: CellConfigurator
+
+  // MARK: - Init
+
+  required init(sections: [Section], configureCell: @escaping CellConfigurator, cellIdentifier: String) {
+    self.unfilteredSections = sections
+    self.sections = sections
     self.configureCell = configureCell
     self.cellIdentifier = cellIdentifier
 
     super.init()
   }
 
+  // MARK: - UICollectionViewDataSource
+
   func numberOfSections(in collectionView: UICollectionView) -> Int {
-    return self.collection.countOfItems
+    sections.count
   }
 
   func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
-    return self.collection[section].countOfItems
+    sections[section].rows.count
   }
 
   func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
-    guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: self.cellIdentifier, for: indexPath) as? HomeCollectionCell else {
+    guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: cellIdentifier, for: indexPath) as? Cell else {
       fatalError("Expecting to dequeue a `CollectionViewCell` from the collectionView")
     }
 
-    let item = self.collection[indexPath]
+    let item = sections[indexPath.section].rows[indexPath.row]
+
     self.configureCell(cell, item)
     return cell
   }
 
-  subscript(indexPath: IndexPath) -> HomeItem {
-    return self.collection[indexPath]
+  // MARK: - Public methods
+  
```

Comment:
```suggestion

```

Reply:
Hmm I don't understand this comment? Are you suggesting to remove a line?

Reply:
OH wait I see! I added extra blank space